suricata/update/main.py

```diff
@@ -353,6 +353,11 @@ def fetch(self, url):
         net_arg = url
         url = url[0] if isinstance(url, tuple) else url
         tmp_filename = self.get_tmp_filename(url)
+        try:
+            with open(tmp_filename, 'wb') as fileobj:
```

Comment:
1. The file was not being created here so this definitely cannot be the place where the error was coming from. Here we are just creating a temporary **file name**.
2.  The error comes on the step where we try to `shutil.copyfile(tmp_fileobj.name, tmp_filename)` i.e. actually opening the file to write the content into it. Read about it here: https://docs.python.org/3/library/shutil.html#shutil.copyfile

Reply:
Yes, I was thinking to explicitly check for the write permissions of dest location in shutil.copyfile. After going through the link, I understood that it itself opens the file and checks that the destination location must be writable. Learned a new thing today. Thanks :)